src/main/java/digital/slovensko/autogram/ui/gui/SigningDialogController.java

```diff
@@ -74,13 +76,19 @@ public void onChangeKeyButtonPressed(ActionEvent event) {
         autogram.pickSigningKeyAndThen(gui::setActiveSigningKey);
     }
 
-    public void refreshSigningKey() {
+    public void refreshSigningKey(boolean shouldLoadDriverFromSettings) {
         mainButton.setDisable(false);
+
         SigningKey key = gui.getActiveSigningKey();
         if (key == null) {
-            mainButton.setText("Vybrať podpisový certifikát");
-            mainButton.getStyleClass().add("autogram-button--secondary");
-            changeKeyButton.setVisible(false);
+            var driver = UserSettings.load().getDriver();
+            if (driver != null && shouldLoadDriverFromSettings) {
```

Comment:
Tuto logiku asi treba aj UX domysliet. Co ak potom kliknem na "zmenit" - ma sa to loadovat ci nema so settings nasledne?

Reply:
Už je to inak spravené. Podľa mňa sa pri prvom výbere má použiť default z nastavení. Pri ďalších by som už normálne dal userovi na výber.